src/ansys/pyensight/core/utils/parts.py

```diff
@@ -122,3 +228,391 @@ def select_parts_by_tag(
+                    )
+                )
+            else:
+                new_emitters.append(
+                    f"ensight.utils.parts._EnSEmitterGrid(ensight, point1={point1}, point2={point2}, point3={point3}, num_points_x={num_points_x}, num_points_y={num_points_y})"
+                )
+        elif emitter_type == self.EMIT_PART:
+            if not parts:
+                raise RuntimeError("part and num_points needed if particle trace emitted from part")
+            for p in parts:
+                if isinstance(self.ensight, ModuleType):
+                    new_emitters.append(
+                        self._EnSEmitterPart(
+                            self.ensight,
+                            part=p,
+                            num_points=num_points,
+                            part_kind=part_distribution_type,
+                        )
+                    )
+                else:
+                    new_emitters.append(
+                        f"ensight.utils.parts._EnSEmitterPart(ensight, part={convert_part(self.ensight ,p)}, num_points={num_points}, part_kind={part_distribution_type})"
+                    )
+        else:
+            raise RuntimeError("No input provided to create the emitters for the particle trace")
+        return new_emitters
+
+    def _create_pathline_part(
+        self,
+        name: str,
+        variable: Union[str, int, "ENS_VAR"],
+        direction: str,
+        source_parts: List[Union[str, int, "ENS_PART"]],
+        pathlines: Optional[bool] = False,
+        emit_time: Optional[float] = None,
+        total_time: Optional[float] = None,
+        delta_time: Optional[float] = None,
+    ) -> Tuple["ENS_PART", "ENS_PART"]:
+        """Private routine to create a pathline part object"""
+        direction_map = {
+            self.PT_POS_TIME: self.ensight.objs.enums.POS_TIME,
+            self.PT_NEG_TIME: self.ensight.objs.enums.NEG_TIME,
+            self.PT_POS_NEG_TIME: self.ensight.objs.enums.POS_NEG_TIME,
+        }
+        converted_source_parts = [convert_part(self.ensight, p) for p in source_parts]
+        idx = self.ensight.objs.enums.PART_PARTICLE_TRACE
+        def_part = self.ensight.objs.core.DEFAULTPARTS[idx]
+        def_part.TYPE = self.ensight.objs.enums.STREAMLINE
+        if pathlines is True:
+            def_part.TYPE = self.ensight.objs.enums.PATHLINE
+        if total_time:
+            def_part.TOTALTIME = total_time
+        if delta_time:
+            def_part.DELTATIME = delta_time
+        if emit_time:
+            def_part.STARTTIME = emit_time
+        def_part.DESCRIPTION = name
+        def_part.VARIABLE = convert_variable(self.ensight, variable)
+        def_part.SURFACERESTRICTED = False
+        def_part.TRACEDIRECTION = direction_map.get(direction)
+        pathline_part = def_part.createpart(sources=converted_source_parts, name=name)[0]
+        return pathline_part, def_part
+
+    def create_particle_trace(
```

Comment:
My only comments here are that I'm wondering if these should be four "simpler" interfaces:  point, line, plane, parts?

Second, what about "POINTS" instead of POINT.  So the caller wants to specify all of the seed points at once (many point emitters).  It might be a handy call.  'create_particle_trace_from_points()' ?

Reply:
@randallfrank  thanks for the suggestions. I split the calls and added the possibility to have multiple points at once